website/docs/r/nsxt_edgegateway_l2_vpn_tunnel.html.markdown

```diff
@@ -0,0 +1,174 @@
+---
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_nsxt_edgegateway_l2_vpn_tunnel"
+sidebar_current: "docs-vcd-resource-nsxt-edgegateway-l2-vpn-tunnel"
+description: |-
+  Provides a resource to manage NSX-T Edge Gateway L2 VPN Tunnel sessions and their configurations.
+---
+
+# vcd\_nsxt\_edgegateway\_l2\_vpn\_tunnel
+
+Supported in provider *v3.11+* and VCD *10.4+* with NSX-T
+
+Provides a resource to manage NSX-T Edge Gateway L2 VPN Tunnel sessions and their configurations.
+<a id="example-usage"></a>
+## Example Usage (Both server and client tunnel sessions connecting two Edge Gateways)
+
+```hcl
+data "vcd_org_vdc" "existing" {
+  name = "existing-vdc"
+}
+
+data "vcd_nsxt_edgegateway" "server-testing" {
+  owner_id = data.vcd_org_vdc.existing.id
+  name     = "server-testing"
+}
+
+data "vcd_nsxt_edgegateway" "client-testing" {
+  owner_id = data.vcd_org_vdc.existing.id
+  name     = "client-testing"
+}
+
+resource "vcd_nsxt_edgegateway_l2_vpn_tunnel" "server-session" {
+  org             = "datacloud"
+  edge_gateway_id = data.vcd_nsxt_edgegateway.server-testing.id
+
+  name        = "server-session"
+  description = "example description"
+
+  session_mode              = "SERVER"
+  enabled                   = true
+  connector_initiation_mode = "ON_DEMAND"
+
+  # must be sub-allocated on the Edge Gateway
+  local_endpoint_ip  = "10.10.50.2"
+  tunnel_interface   = "192.168.0.1/24"
+  remote_endpoint_ip = "1.2.2.3"
+
+  stretched_network {
+    network_id = data.vcd_network_routed_v2.test_network_server.id
+  }
+
+  pre_shared_key = "secret_passphrase"
+}
+
+resource "vcd_nsxt_edgegateway_l2_vpn_tunnel" "client-session" {
+  org = "datacloud"
+
+  # Note that this is a different edge gateway, as one edge gateway
```

Comment:
Please use same capitalization of edge gateway in the same code sample :) (See line 43). BTW, it's fine to leave lower case in such comments. Please double check other samples for consistency as well.

Reply:
Agree, still changed everything to `Edge Gateway` for consistency, thanks